packages/mds-metrics-sheet/metrics-log-utils.ts

```diff
@@ -0,0 +1,191 @@
+export function percent(a: number, total: number) {
+  return Math.round(((total - a) / total) * 10000) / 10000
+}
+
+export const mapProviderToPayload = (provider: VehicleCountRow, last: LastDayStatsResponse) => {
+  const dateOptions = { timeZone: 'America/Los_Angeles', day: '2-digit', month: '2-digit', year: 'numeric' }
+  const timeOptions = { timeZone: 'America/Los_Angeles', hour12: false, hour: '2-digit', minute: '2-digit' }
+  const d = new Date()
+  let [enters, leaves, starts, ends, start_sla, end_sla, telems, telem_sla] = [0, 0, 0, 0, 0, 0, 0, 0]
+  let event_counts = { service_start: 0, provider_drop_off: 0, trip_start: 0, trip_end: 0 }
+  let status_counts = {
+    available: 0,
+    unavailable: 0,
+    reserved: 0,
+    trip: 0,
+    removed: 0,
+    inactive: 0,
+    elsewhere: 0
+  }
+  const { event_counts_last_24h, late_event_counts_last_24h, late_telemetry_counts_last_24h } = last[
+    provider.provider_id
+  ]
+  if (event_counts_last_24h) {
+    event_counts = event_counts_last_24h
+    status_counts = eventCountsToStatusCounts(event_counts_last_24h)
+    starts = event_counts_last_24h.trip_start || 0
+    ends = event_counts_last_24h.trip_end || 0
+    enters = event_counts_last_24h.trip_enter || 0
+    leaves = event_counts_last_24h.trip_leave || 0
+    telems = last[provider.provider_id].telemetry_counts_last_24h || 0
+    if (late_telemetry_counts_last_24h !== undefined && late_telemetry_counts_last_24h !== null) {
+      telem_sla = telems ? percent(late_telemetry_counts_last_24h, telems) : 0
+    }
+    if (late_event_counts_last_24h !== undefined && late_event_counts_last_24h !== null) {
+      start_sla = starts ? percent(late_event_counts_last_24h.trip_start, starts) : 0
+      end_sla = ends ? percent(late_event_counts_last_24h.trip_end, ends) : 0
+    }
+  }
+  return {
+    date: `${d.toLocaleDateString('en-US', dateOptions)} ${d.toLocaleTimeString('en-US', timeOptions)}`,
+    name: provider.provider,
+    registered: provider.count || 0,
+    deployed:
+      sum([provider.status.available, provider.status.unavailable, provider.status.trip, provider.status.reserved]) ||
+      0,
+    validtrips: 'tbd', // Placeholder for next day valid trip analysis
+    trips: last[provider.provider_id].trips_last_24h || 0,
+    servicestart: event_counts.service_start || 0,
+    providerdropoff: event_counts.provider_drop_off || 0,
+    tripstart: starts,
+    tripend: ends,
+    tripenter: enters,
+    tripleave: leaves,
+    telemetry: telems,
+    telemetrysla: telem_sla,
+    tripstartsla: start_sla,
+    tripendsla: end_sla,
+    available: status_counts.available,
+    unavailable: status_counts.unavailable,
+    reserved: status_counts.reserved,
+    trip: status_counts.trip,
+    removed: status_counts.removed,
+    inactive: status_counts.inactive,
+    elsewhere: status_counts.elsewhere
+  }
+}
+
+export const getProviderMetrics = async (iter: number): Promise<ProviderMetrics> => {
```

Comment:
This has been reworked. The idea now is to delegate handling exceptions/logging to the API call helpers. The helpers will return `null` if they encounter an error, so we use that as a signal to retry.

Reply:
This is perhaps outside the scope of this particular PR, but I'm not a huge fan of just catching errors and logging them. If, say, the improper credentials have been supplied, how would we know? AFAICT this thing will just retry and suffer in silence.

Reply:
All the API call helper methods have catch blocks, so they will log any errors.

I wanted to normalize the API calls so they only ever return the payload or null, it makes it a good deal easier to reason about + implement the retry logic.

Reply:
The retry logic was included b/c some of the services we call live(d) on AWS Lambda, so sometimes we'd hit timeout when we have to cold start. Perhaps this is a non-issue now that we've migrated most services to k8s.

Reply:
@janedotx I'm not sure what you're getting at... Bar adding complex logic to handle every possible failure case by deconstructing the response, I don't see how else we would handle the error besides catching it and logging it... Historically, the mds-daily endpoints have had issues with timing out, hence adding the retry and log mechanism, which is likely not a problem within the kubernetes architecture for mds-core (no more API Gateway timeouts). I'd rather have it logged, send me an alert via AWS cloudwatch, and retry than just explode and not keep trying... 

Keep in mind this is not meant to be a long-term supported package. We're hoping to fully deprecate and purge it in Q4.